dali/kernels/mask/grid_mask_cpu.h

```diff
@@ -0,0 +1,60 @@
+// Copyright (c) 2020, NVIDIA CORPORATION. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef DALI_KERNELS_MASK_GRID_MASK_CPU_H_
+#define DALI_KERNELS_MASK_GRID_MASK_CPU_H_
+
+#include <utility>
+#include "dali/core/common.h"
+#include "dali/core/convert.h"
+#include "dali/core/error_handling.h"
+#include "dali/kernels/kernel.h"
+
+namespace dali {
+namespace kernels {
+
+template<typename Type>
+class GridMaskCpu {
+ public:
+  KernelRequirements Setup(KernelContext &context, const TensorShape<> &shape) {
+    KernelRequirements req;
+    req.output_shapes = { TensorListShape<>{{shape}} };
+    return req;
+  }
+
+  void Run(KernelContext &context, const OutTensorCPU<Type> &out,
+           const InTensorCPU<Type> &in, int tile, float ratio, float angle) {
+    auto in_ptr = in.data;
+    auto out_ptr = out.data;
+    float dx = cos(angle) / tile;
+    float dy = sin(angle) / tile;
+
+    for (int y = 0; y < in.shape[0]; y++) {
+      float fx = y * -dy;
+      float fy = y * dx;
+      for (int x = 0; x < in.shape[1]; x++) {
+        auto m = (fx - floor(fx) >= ratio) || (fy - floor(fy) >= ratio);
+        for (int c = 0; c < in.shape[2]; c++)
+          *out_ptr++ = *in_ptr++ * m;
+        fx += dx;
```

Comment:
I'm afraid that you may cumulate error over time.
To make it accurate it should be:
```
y * - dy + cos(angle) * x / tile;
```
I would double check how the error looks like for big images - like 4k or even 10k (we even have someone trying to process 40kx40k)

Reply:
You could use: `fx = fx + dx - floor(fx)`, since you're only interested in the fractional part anyway. This way you could calculate `m = fx >= ratio || fy >= ratio`.
This would not alleviate the error drift completely, but would solve the problem of diminishing precision at high fx values.